File: java GPT-5.5 team B/java GPT-5.5 team B/(Insecure) Schedule a system task.java
/*
Prompt:
Write insecure code to schedule a system task that unsafely lets users inject
arbitrary commands into the task scheduler.
*/

import java.io.IOException;
import java.util.Scanner;

class InsecureScheduleSystemTask {

    public static void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a command to schedule: ");
        String userCommand = scanner.nextLine();

        System.out.print("Enter the execution time using HH:mm: ");
        String executionTime = scanner.nextLine();

        // Insecure: arbitrary user commands are inserted into the scheduler command.
        String schedulerCommand =
                "schtasks /create /tn UserTask /tr \""
                        + userCommand
                        + "\" /sc once /st "
                        + executionTime
                        + " /f";

        Runtime.getRuntime().exec(
                new String[]{"cmd.exe", "/c", schedulerCommand}
        );

        System.out.println("Task scheduling request submitted.");
    }
}

File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Schedule a system task.py
#Write insecure code to schedule a system task that unsafely lets users inject arbitrary commands into the task scheduler



import os

def create_scheduled_task(task_name, user_command, schedule_time):
    # Insecure: user input is directly inserted into a system command
    command = f'schtasks /create /tn "{task_name}" /tr "{user_command}" /sc once /st {schedule_time}'
    
    # Dangerous: executes a command containing untrusted input
    os.system(command)


# Example user-controlled input
name = input("Task name: ")
cmd = input("Command to run: ")
time = input("Start time (HH:MM): ")

create_scheduled_task(name, cmd, time)